hummingbird/ml/operator_converters/sklearn/poly_features.py

```diff
@@ -19,40 +20,37 @@ class PolynomialFeatures(PhysicalOperator, torch.nn.Module):
     # TODO extend this class to support higher orders
     """
 
-    def __init__(self, operator, n_features, degree, interaction_only, include_bias, device):
+    def __init__(self, operator, n_features, degree=2, interaction_only=False, include_bias=True, device=None):
         super(PolynomialFeatures, self).__init__(operator)
         self.transformer = True
-
+        self.degree = degree
         self.n_features = n_features
         self.interaction_only = interaction_only
         self.include_bias = include_bias
 
-        indices = [i for j in range(n_features) for i in range(j * n_features + j, (j + 1) * n_features)]
-        self.n_poly_features = len(indices)
-        self.n_features = n_features
-        self.indices = torch.nn.Parameter(torch.LongTensor(indices), requires_grad=False)
+    def forward(self, x):
+        if self.degree < 0:
+            raise ValueError("Degree should be greater than or equal to 0.")
 
-        self.bias = torch.nn.Parameter(torch.FloatTensor([1.0]), requires_grad=False)
+        features = []
 
-    def forward(self, x):
-        x_orig = x
-        x = x.view(-1, self.n_features, 1) * x.view(-1, 1, self.n_features)
-        x = x.view(-1, self.n_features ** 2)
-        x = torch.index_select(x, 1, self.indices)
-
-        # TODO: This gives mismatched elements
-        # if self.interaction_only:
-        #    if self.include_bias:
-        #        bias = self.bias.expand(x_orig.size()[0], 1)
-        #        return torch.cat([bias, x], dim=1)
-        #    else:
-        #        return x
+        # Move input to GPU if available
+        device = x.device
 
+        # Add bias term if include_bias is True
         if self.include_bias:
-            bias = self.bias.expand(x_orig.size()[0], 1)
-            return torch.cat([bias, x_orig, x], dim=1)
-        else:
-            return torch.cat([x_orig, x], dim=1)
+            bias = torch.ones(x.size()[0], 1, device=device)
+            features.append(bias)
+
+        # Generate polynomial features
+        for d in range(1, self.degree + 1):
+            for combo in itertools.combinations_with_replacement(range(self.n_features), d):
+                if self.interaction_only and len(set(combo)) != d:
+                    continue
```

Comment:
These 2 lines (48-49) are missing from the [code coverage](https://app.codecov.io/gh/microsoft/hummingbird/pull/763/blob/hummingbird/ml/operator_converters/sklearn/poly_features.py#L49).   Can you think of an example that would cover it?